### PR TITLE
Implement EditBookScreen logic and UI for editing and deleting books

### PR DESCRIPTION
**Pull request by LeniCadorinEpfl**
This PR introduces the `EditBookScreen`, which allows users to edit book details and delete books.

**Key components include:**

**EditBookScreen UI Implementation:** A complete layout for editing book information, including title, genre, author, description, genre, rating, photo, and language.

EditBookScreenUI

**Core Logic Integration:** Utilized the `BooksFirestoreRepository` for data management, with functions to update and delete books, along with validation checks for required fields.

**Error Handling:** Added Toast notifications for handling validation errors and update failures, ensuring a good user experience.

**Tests for EditBookScreen:** Added tests in EditBookScreenTest.kt to ensure correct functionality

**Checklist:**
- [ ] CI passes
- [x] Structure is correct
- [x] Tests for the added code are complete

### CODE REVIEW

**Review by TheoSchlaeppi — CHANGES_REQUESTED** [review]
Here are my thoughts and suggestions:

1. **EditBookScreen UI Implementation**:
   - The layout for editing book information is comprehensive and includes fields for title, genre, author, description, rating, photo, and language.
   - Consider verifying the correct initial values in your tests to ensure the UI displays the expected data when loading the screen.

2. **Core Logic Integration**:
   - The integration with `BooksFirestoreRepository` for data management is well done. Functions for updating and deleting books are clearly defined.
   - It would be beneficial to add a test to confirm that the save button correctly calls the repository's `updateBook` function with the right parameters.

3. **Error Handling**:
   - The inclusion of Toast notifications for validation errors and update failures enhances the user experience. Ensure that error messages are clear and helpful to the user.
   - Additionally, consider adding tests for scenarios where fields are left blank to verify that appropriate error messages are displayed.

4. **Typos**
   - There are some typos that i have flag with comments on the code.

### Overall Impression

The implementation of `EditBookScreen` is well-structured and addresses key functionalities for editing book details.I appreciate how you structured the code, making it easy to integrate additional features in the future.

Thank you for your work on this! Looking forward to seeing the improvements.

**Comment by LeniCadorinEpfl** [comment]
Thank you for the review. I have taken your suggestions into account and removed the different typo.
For the part:
> Consider verifying the correct initial values in your tests to ensure the UI displays the expected data when loading the screen.
These test are already present in my code, the test name is 'inputsHaveInitialValue`.

File: app/src/main/java/com/android/bookswap/ui/books/edit/EditBookScreen.kt [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+                              description = description,
+                              ratingStr = rating,
+                              photo = photo,
+                              bookLanguageStr = language,
+                              isbn = book.isbn,
+                              genres = genres)
+
+                      booksRepository.updateBook(
+                          updatedBook!!,
+                          OnSucess = { navigationActions.goBack() },
```
[the remark this answers is not in the document]
Comment (LeniCadorinEpfl) [reply app/src/main/java/com/android/bookswap/ui/books/edit/EditBookScreen.kt:305]: This is from Matias's code, but I can actually change it.

**Review by TheoSchlaeppi — APPROVED** [review]
Looks good for me !

**Pull request merged**